Sources/Defaults/Defaults+AnySerializable.swift

```diff
@@ -0,0 +1,257 @@
+			return hasher.combine(value)
+		case let value as String:
+			return hasher.combine(value)
+		case let value as [AnyHashable: AnyHashable]:
+			return hasher.combine(value)
+		case let value as [AnyHashable]:
+			return hasher.combine(value)
+		default:
+			break
+		}
+	}
+}
+
+extension Defaults.AnySerializable: Equatable {
+	public static func == (lhs: Defaults.AnySerializable, rhs: Defaults.AnySerializable) -> Bool {
+		switch (lhs.value, rhs.value) {
+		case let (lhs as Data, rhs as Data):
+			return lhs == rhs
+		case let (lhs as Date, rhs as Date):
+			return lhs == rhs
+		case let (lhs as Bool, rhs as Bool):
+			return lhs == rhs
+		case let (lhs as UInt8, rhs as UInt8):
+			return lhs == rhs
+		case let (lhs as Int8, rhs as Int8):
+			return lhs == rhs
+		case let (lhs as UInt16, rhs as UInt16):
+			return lhs == rhs
+		case let (lhs as Int16, rhs as Int16):
+			return lhs == rhs
+		case let (lhs as UInt32, rhs as UInt32):
+			return lhs == rhs
+		case let (lhs as Int32, rhs as Int32):
+			return lhs == rhs
+		case let (lhs as UInt64, rhs as UInt64):
+			return lhs == rhs
+		case let (lhs as Int64, rhs as Int64):
+			return lhs == rhs
+		case let (lhs as UInt, rhs as UInt):
+			return lhs == rhs
+		case let (lhs as Int, rhs as Int):
+			return lhs == rhs
+		case let (lhs as Float, rhs as Float):
+			return lhs == rhs
+		case let (lhs as Double, rhs as Double):
+			return lhs == rhs
+		case let (lhs as CGFloat, rhs as CGFloat):
+			return lhs == rhs
+		case let (lhs as String, rhs as String):
+			return lhs == rhs
+		case let (lhs as [AnyHashable: Any], rhs as [AnyHashable: Any]):
+			return lhs.toDictionary() == rhs.toDictionary()
+		case let (lhs as [Any], rhs as [Any]):
+			return lhs.toSequence() == rhs.toSequence()
+		default:
+			return false
+		}
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByStringLiteral {
+	public init(stringLiteral value: String) {
+		self.init(value: value)
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByNilLiteral {
+	public init(nilLiteral _: ()) {
+		self.init(value: nil as Any?)
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByBooleanLiteral {
+	public init(booleanLiteral value: Bool) {
+		self.init(value: value)
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByIntegerLiteral {
+	public init(integerLiteral value: Int) {
+		self.init(value: value)
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByFloatLiteral {
+	public init(floatLiteral value: Double) {
+		self.init(value: value)
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByArrayLiteral {
+	public init(arrayLiteral elements: Any...) {
+		self.init(value: elements)
+	}
+}
+
+extension Defaults.AnySerializable: ExpressibleByDictionaryLiteral {
+	public init(dictionaryLiteral elements: (AnyHashable, Any)...) {
+		self.init(value: [AnyHashable: Any](uniqueKeysWithValues: elements))
+	}
+}
+
+extension Defaults.AnySerializable: _DefaultsOptionalType {
+	public var isNil: Bool { value is Void }
```

Comment:
This could use a doc comment. It's not immediately clear why it's `nil` when it's `Void`.